src/components/Amalgamation/AmalgamatingBusinesses.vue

```diff
@@ -153,92 +164,115 @@ import BusinessTable from '@/components/Amalgamation/BusinessTable.vue'
 
-    // Get the amalgamating business information, mailing address, and email if in LEAR.
-    // Otherwise, return the businesslookup object.
-    const data = await Promise.all([
-      LegalServices.fetchBusinessInfo(businessLookup.identifier),
-      AuthServices.fetchAuthInfo(businessLookup.identifier),
-      LegalServices.fetchAddresses(businessLookup.identifier)
-    ]).catch((error) => {
-      return error
-    })
-
-    if (data.length === 3) {
-      business = data[0].data?.business
-      business.businessContact = data[1].contacts[0]
-      business.officeAddress = data[2]
+      return
     }
 
-    // If the business is not null (LEAR Entity), create from it a TING business following the interface.
-    // If the amalgamating businesses array is not empty, check if identifier already exists.
-    // If identifier already exists, don't add the business to the array.
-    if (business) {
-      const amalgamatingBusinesses = this.getAmalgamatingBusinesses
+    // Check for Legal API fetch issues.
+    if (!businessInfo || !addresses || !firstFiling) {
+      this.snackbarText = 'Unable to add that business.'
+      this.snackbar = true
+      return
+    }
 
-      const tingBusiness = {
-        type: 'lear',
-        role: AmlRoles.AMALGAMATING,
-        identifier: business.identifier,
-        name: business.legalName,
-        email: business.businessContact.email,
-        legalType: business.legalType,
-        address: business.officeAddress.registeredOffice.mailingAddress,
-        goodStanding: business.goodStanding
-      } as AmalgamatingBusinessIF
-
-      if (!amalgamatingBusinesses.find((b: any) => b.identifier === business.identifier)) {
-        amalgamatingBusinesses.push(tingBusiness)
-
-        // Set the new amalgamated businesses array in the store.
-        this.setAmalgamatingBusinesses(amalgamatingBusinesses)
-      }
+    // Verify that identifier doesn't already exist.
+    if (this.getAmalgamatingBusinesses.find((b: any) => b.identifier === businessInfo.identifier)) {
+      this.snackbarText = 'Business is already in table.'
+      this.snackbar = true
+      return
     }
 
-    // Close the "Add an Amalgamating Business" Panel.
+    // If there is a state filing and restoration expiry date isn't in the past and the state filing is a
+    // limited restoration or limited restoration extension, then this business is in limited restoration.
+    const isLimitedRestoration = async (): Promise<boolean> => {
+      // check for no state filing
+      if (!businessInfo.stateFiling) return false
+      // check for expired restoration
+      if (this.getCurrentDate > businessInfo.restorationExpiryDate) return false
+      // fetch state filing
+      const stateFiling = await LegalServices.fetchFiling(businessInfo.stateFiling)
+      return (
+        stateFiling.restoration.type === RestorationTypes.LIMITED ||
+        stateFiling.restoration.type === RestorationTypes.LTD_EXTEND
+      )
+    }
```

Comment:
One advantage of having this nested function is that `businessInfo` is already in scope so I don't have to pass it in as a formal parameter.